src/coreclr/jit/importer.cpp

```diff
@@ -7883,6 +7883,16 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                                || (impStackTop().val->TypeGet() == TYP_BYREF)
 #endif
                         ;
+#ifdef TARGET_AMD64
+                    // If AVX512 is present and we are not checking for overflow, we do not need
+                    // a large node. In this case, we will not fallback to a helper function but
+                    // will use the intrinsic instead. Hence setting the callNode to false to
+                    // avoid generating a large node.
+                    if (callNode && compOpportunisticallyDependsOn(InstructionSet_AVX512F) && !ovfl)
+                    {
+                        callNode = false;
+                    }
+#endif // TARGET_AMD64
```

Comment:
Is this correctly handling `TARGET_X86`?

That is, `callNode` is currently true for `long->float/double` and `ulong->float/double`. This can only be done on `TARGET_AMD64` since `vcvttusi2ss/sd r64` is 64-bit only.

However, `callNode` will also be true for `uint->float/double` which is 32-bit or 64-bit.

There's also the case where `callNode` is being set to `true` for `TYP_BYREF` and so we may need to not do this optimization in that case (at least until the general todo comment above is handled).

Reply:
Changed this to generate a small node in cases of long/ulong -> floating point types. And moved the compOpportunisticallyDependsOn check at the last position.